packages/libs/user-datasets/src/lib/Service/api.ts

```diff
@@ -0,0 +1,230 @@
+import { zipWith } from 'lodash';
+
+import {
+  arrayOf,
+  number,
+  objectOf,
+  record,
+} from '@veupathdb/wdk-client/lib/Utils/Json';
+
+import {
+  createJsonRequest,
+  FetchClientWithCredentials,
+  ioTransformer,
+} from '@veupathdb/http-utils';
+
+import {
+  userDataset,
+  UserDatasetMeta,
+  NewUserDatasetMeta,
+  NewUserDataset,
+  userDatasetDetails,
+  userQuotaMetadata,
+  userDatasetFileListing,
+  datasetIdType,
+} from '../Utils/types';
+
+import { array } from 'io-ts';
+import { submitAsForm } from '@veupathdb/wdk-client/lib/Utils/FormSubmitter';
+
+export const VDI_SERVICE_BASE_URL = 'http://localhost:8080';
+const VDI_SERVICE = '/vdi-datasets';
+
+const userIdsByEmailDecoder = record({
+  results: arrayOf(objectOf(number)),
+});
+
+export class UserDatasetApi extends FetchClientWithCredentials {
```

Comment:
I see that you are using class field functions, rather than class methods. What's the reason for this? Maybe add a comment explaining why.

Reply:
Hmm, it wasn't a deliberate choice- more of a reflex. I don't suspect we'll have many instances of this API floating around, so it's not a huge issue for the functions to not be on the prototype. But it's probably worth switching them to class methods since I can't think of a need for them to be field functions. Am I missing something?